File: C++/Medium/WrongAnswer/find-missing-observations_3.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        
        int m = rolls.size();
        int sum = accumulate(rolls.begin(), rolls.end(), 0);
        int total = mean * (m + n);
        int missing = total - sum;
        if (missing < n || missing > 6 * n) {
            return {};
        }
        vector<int> result(n);
        for (int i = 0; i < n; ++i) {
            result[i] = missing / n;
            missing -= result[i];
            --n;
        }
        for (int i = 0; i < result.size(); ++i) {
            if (result[i] == 0) {
                result[i] = 1;
                --missing;
            }
        }
        for (int i = result.size() - 1; i >= 0 && missing > 0; --i) {
            if (result[i] < 6) {
                ++result[i];
                --missing;
            }
        }
        return result;
    }
};

File: Java/Medium/WrongAnswer/find-missing-observations_3.java
class Solution {
    public int[] missingRolls(int[] rolls, int mean, int n) {
        
        int sum = 0;
        for (int i = 0; i < rolls.length; i++) {
            sum += rolls[i];
        }
        sum += mean * (n + rolls.length);
        if (sum % (n + rolls.length) != 0) {
            return new int[0];
        }
        int avg = sum / (n + rolls.length);
        int[] result = new int[n];
        for (int i = 0; i < n; i++) {
            result[i] = avg;
        }
        int diff = avg * n;
        for (int i = 0; i < rolls.length; i++) {
            diff -= rolls[i];
        }
        for (int i = 0; i < n; i++) {
            if (diff >= 5) {
                result[i] += 5;
                diff -= 5;
            } else {
                result[i] += diff;
                break;
            }
        }
        return result;
    }
}